### DISCUSSION TITLE
Are there any problems that may arise from using both Provider and Riverpod (the dart only package) in a project?

### DISCUSSION THREAD

**Discussion by joaopegoraro** [rrousselGit/riverpod#2674] [Q&A] [answered]
I prefer the original Provider way of dealing with UI, but I really like Riverpod as a dependency injection solution. Is there any problem in using both? For example:

I have a View Model (a ChangeNotifier class) that depends on an ApiService class. So, I instantiate the ViewModel the normal way with Provider, but I still need some way to access the ProviderRef there so I can pass the ApiService to the ViewModel constructor. So the way to achieve that, is to pass the ProviderContainer from Riverpod as a Provider, like so:
```
void main() {
  final container = riverpod.ProviderContainer();
  runApp(
    Provider.value(
      value: container,
      child: const MyApp(),
    ),
  );
}
```
And to access it:
```
 @override
  Widget build(BuildContext context) {
    return ChangeNotifierProvider(
      create: (context) {
        final apiService = context.read<ProviderContainer>().read(apiServiceProvider);
        return ViewModel(apiService);
      },
      child: Consumer(builder: (...)),
    );
  }
```
Other than looking weird (nothing that can't be fixed by some extension methods), are there any problems with this approach?

**Comment by rrousselGit** [comment, marked as answer]
Using both is fine.

**Reply by joaopegoraro** [reply]
Nice! So there is no problem with passing ProviderContainer to Provider.value? Like any leak or something?

**Reply by rrousselGit** [reply]
Well it's discouraged. But it's going to work.
Make a consumer instead though
